client.go

```diff
@@ -1125,6 +1125,21 @@ func (c *Client) GetClient() *http.Client {
 	return c.httpClient
 }
 
+// Clone returns a clone of the original client.
+//
+// Be carefull when using this function:
+// - Interface values are not deeply cloned. Thus, both the original and the clone will use the
+// same value.
+// - This function is not safe for concurrent use. You should only use this when you are sure that
+// the client is not being used by any other goroutine.
+//
+// Since v2.12.0
+func (c *Client) Clone() *Client {
+	// dereference the pointer and copy the value
+	cc := *c
+	return &cc
```

Comment:
@fabiante Just a while ago, I validated it. We need to reinitialize the lock after the clone.

The complete clone method is as follows. Can you please update the PR?
```go
func (c *Client) Clone() *Client {
	// dereference the pointer and copy the value
	cc := *c
	cc.afterResponseLock = &sync.RWMutex{}
	cc.udBeforeRequestLock = &sync.RWMutex{}
	return &cc
}
```

Reply:
I also think this makes more sense. Linters my still give a warning / hint about the dereferencing but the code should obiously show that we deal with this appropriately.